lib/rules/jsx-no-literals.js

```diff
@@ -182,9 +182,8 @@ const elementOverrides = {
   },
 };
 
-/** @type {import('eslint').Rule.RuleModule} */
 module.exports = {
-  meta: {
+  meta: /** @type {import('eslint').Rule.RuleModule["meta"]} */ ({
```

Comment:
this seems worse, since it's checking less of the module - why is it needed?

Reply:
There seems to be a bug in the generated type declaration otherwise.

I did contemplate whether one should maybe use the latest TS release for generating the type declarations, as bugs like that won't get fixed in `^3.9.9`

Reply:
I would certainly expect that - at this point i don't even care much about someone using TS < 5.

Reply:
Would it be possible to update the TypeScript version for the entire project? I guess it's pinned to 3.x because of the version of the typescript-eslint-parser?

Reply:
Yes, that's why.

It would indeed be great to update the dev deps range to allow `^3.9.9 || ^5.6.2`, and then modify CI so it installs the appropriate typescript version as needed! That'd work well as a separate PR.